docs/development/extensions-contrib/spectator-histogram.md

```diff
@@ -0,0 +1,453 @@
+          "regionIsoCode",
+          "metroCode",
+          "countryIsoCode",
+          "regionName"
+        ]
+      },
+      "metricsSpec": [
+        { "name": "count", "type": "count" },
+        { "name": "sum_added", "type": "longSum", "fieldName": "added" },
+        {
+          "name": "hist_added",
+          "type": "spectatorHistogram",
+          "fieldName": "added"
+        }
+      ]
+    },
+    "tuningConfig": {
+      "type": "index_parallel",
+      "partitionsSpec": { "type": "hashed" },
+      "forceGuaranteedRollup": true
+    }
+  }
+}
+```
+
+### Example Query
+Example query using the sample wikipedia dataset:
+```json
+{
+  "queryType": "timeseries",
+  "dataSource": {
+    "type": "table",
+    "name": "wikipedia"
+  },
+  "intervals": {
+    "type": "intervals",
+    "intervals": [
+      "0000-01-01/9999-12-31"
+    ]
+  },
+  "granularity": {
+    "type": "all"
+  },
+  "aggregations": [
+    {
+      "type": "spectatorHistogram",
+      "name": "histogram_added",
+      "fieldName": "added"
+    }
+  ],
+  "postAggregations": [
+    {
+      "type": "percentileSpectatorHistogram",
+      "name": "medianAdded",
+      "field": {
+        "type": "fieldAccess",
+        "fieldName": "histogram_added"
+      },
+      "percentile": "50.0"
+    }
+  ]
+}
+```
+Results in
+```json
+[
+  {
+    "result": {
+      "histogram_added": {
+        "0": 11096, "1": 632, "2": 297, "3": 187, "4": 322, "5": 161,
+        "6": 174, "7": 127, "8": 125, "9": 162, "10": 123, "11": 106,
+        "12": 95, "13": 104, "14": 95, "15": 588, "16": 540, "17": 690,
+        "18": 719, "19": 478, "20": 288, "21": 250, "22": 219, "23": 224,
+        "24": 737, "25": 424, "26": 343, "27": 266, "28": 232, "29": 217,
+        "30": 171, "31": 164, "32": 161, "33": 530, "34": 339, "35": 236,
+        "36": 181, "37": 152, "38": 113, "39": 128, "40": 80, "41": 75,
+        "42": 289, "43": 145, "44": 138, "45": 83, "46": 45, "47": 46,
+        "48": 64, "49": 65, "50": 71, "51": 421, "52": 525, "53": 59,
+        "54": 31, "55": 35, "56": 8, "57": 10, "58": 5, "59": 4, "60": 11,
+        "61": 10, "62": 5, "63": 2, "64": 2, "65": 1, "67": 1, "68": 1,
+        "69": 1, "70": 1, "71": 1, "78": 2
+      },
+      "medianAdded": 4.013975155279504
+    },
+    "timestamp": "2016-06-27T00:00:00.000Z"
+  }
+]
+```
+
+### Histogram Bucket Boundaries
+These are the upper bounds of each bucket index. There are 276 buckets.
+The first bucket index is 0 and the last bucket index is 275.
+As you can see the bucket widths increase as the bucket index increases. This leads to a greater absolute error for larger values, but maintains a relative error of rough percentage across the number range.
+i.e the maximum error at value 10 is 0 since the bucket width is 1. But for a value of 16,000,000,000 the bucket width is 1,431,655,768 giving an error of up to ~8.9%. In practice, the observed error of computed percentiles is in the range (0.1%, 3%).
```

Comment:
```suggestion
For example, the maximum error at value 10 is zero since the bucket width is 1 (the difference of 11-10). For a value of 16,000,000,000, the bucket width is 1,431,655,768 giving an error of up to ~8.9%. In practice, the observed error of computed percentiles is in the range of (0.1%, 3%).
```
Not sure if it's 11-10 or 10-9. Also it's not clear how you get 8.9%. Consider explaining in more detail, or linking to relevant docs.